esmvaltool/diag_scripts/extreme_events/make_timeseries_plot.r

```diff
@@ -0,0 +1,580 @@
+        )
+
+        # Fieldmean results
+        cdo("fldmean",
+          input = paste0(land, "/detrend_standard_", o),  # nolint
+          output = paste0(land, "/detrend_std_fldm_", o), options = "-O"  # nolint
+        )
+
+        # Copy obs file to plot
+        n <- n + 1
+        file.copy(
+          paste0(land, "/detrend_std_fldm_", o),  # nolint
+          paste0(
+            plot_dir, "/", idx, "_",
+            modelsandobssplit[obs_order[n]],
+            "_for_timeseries.nc"
+          )
+        )
+      }
+    }
+  }
+
+  # ABSOLUTE VALUES ####
+  # Non-normalized values fieldmeans
+  if (!normalize) {
+    file_string_models <- ""
+    m <- models[1]
+    for (m in models) {
+      print(m)
+      # Fieldmean results
+      cdo("fldmean",
+        input = paste0(land, "/", m),
+        output = paste0(land, "/fldm_", m), options = "-O"  # nolint
+      )
+
+      ## add the preprocessed file to the filestring
+      file_string_models <- paste(file_string_models, land,
+        "/fldm_", m, " ", sep = "")  # nolint
+    }
+    # Find model ensemble mean
+    cdo("ensmean",
+      input = file_string_models,
+      output = paste0(plot_dir, "/", idx, "_ensmean_for_timeseries.nc"),
+      options = "-O"
+    )
+
+    # Find ensemble 25th percentile
+    cdo("enspctl",
+      args = "25", input = file_string_models,
+      output = paste0(plot_dir, "/", idx, "_25enspctl_for_timeseries.nc"),
+      options = "-O"
+    )
+
+    # Find ensemble 75th percentile
+    cdo("enspctl",
+      args = "75", input = file_string_models,
+      output = paste0(plot_dir, "/", idx, "_75enspctl_for_timeseries.nc"),
+      options = "-O"
+    )
+
+    ## Extracting only the observation files
+    obs_order <- which(modelsandobssplit %in% obs_list)
+    obs <- modelsandobs[obs_order]
+
+    print("These are the observations:")
+    print(obs)
+    n <- 0
+    for (o in obs) {
+      print(o)
+      # Fieldmean results
+      cdo("fldmean",
+        input = paste0(land, "/", o),
+        output = paste0(land, "/fldm_", o), options = "-O"  # nolint
+      )
+      # Copy obs file to plot
+      n <- n + 1
+      file.copy(
+        paste0(land, "/fldm_", o),  # nolint
+        paste0(
+          plot_dir, "/", idx, "_",
+          modelsandobssplit[obs_order[n]], "_for_timeseries.nc"
+        )
+      )
+    }
+  }
+}
+
+#
+#
+# Method to plot the time series plot
+# of single idx for already preprocessed data
+# yearly data is assumed
+# @param path - path to directory containing ensemble mean
+# and percentile data.
+# @param idx name of index to be processed
+# @param start_yr start year for data to be used to convert
+# values from days after start year format to
+# year.
+#
+
+timeseries_plot <- function(plot_dir = "./plot", idx = "tn10pETCCDI_yr",
+                            obs_list, start_yr = 2006, end_yr = 2010,
+                            normalize = FALSE) {
+  ## Loading ncdf4 library
+  library(ncdf4)
```

Comment:
Should be done at the top of the file

Reply:
Ok, I moved all `library()` calls to the top of the main driver `extreme_events.R`